# README for realignment of the HGSV trio sequences to hs37d5
PCR-free Illumina high coverage (75X) sequences of three trio: CHS, PUR, YRI, that have been produced by the HGSVC have been down sampled to ~30X and realigned to hs37d5 

## hs37d5 is used as the alignment reference, which is dowoloaded from [the 1KGP ftp site](ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/), and then indexed by bwa and picard
```
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/*
bwa index hs37d5.fa
java -jar picard.jar CreateSequenceDictionary R=reference.fa O=reference.dict
```

working directory on erisone: 
```
/data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_GRCh38
```

## Step1: extract fastqs from HGSV alignment in cram format:
```
bsub -q medium -J HG00512 -o HG00512.log -sla miket_sc "samtools fastq -1 HG00512.1.fq -2 HG00512.2.fq HG00512.alt_bwamem_GRCh38DH.20150715.CHS.high_coverage.cram"
bsub -q medium -J HG00513 -o HG00513.log -sla miket_sc "samtools fastq -1 HG00513.1.fq -2 HG00513.2.fq HG00513.alt_bwamem_GRCh38DH.20150715.CHS.high_coverage.cram"
bsub -q medium -J HG00514 -o HG00514.log -sla miket_sc "samtools fastq -1 HG00514.1.fq -2 HG00514.2.fq HG00514.alt_bwamem_GRCh38DH.20150715.CHS.high_coverage.cram"
bsub -q medium -J HG00731 -o HG00731.log -sla miket_sc "samtools fastq -1 HG00731.1.fq -2 HG00731.2.fq HG00731.alt_bwamem_GRCh38DH.20150715.PUR.high_coverage.cram"
bsub -q medium -J HG00732 -o HG00732.log -sla miket_sc "samtools fastq -1 HG00732.1.fq -2 HG00732.2.fq HG00732.alt_bwamem_GRCh38DH.20150715.PUR.high_coverage.cram"
bsub -q medium -J HG00733 -o HG00733.log -sla miket_sc "samtools fastq -1 HG00733.1.fq -2 HG00733.2.fq HG00733.alt_bwamem_GRCh38DH.20150715.PUR.high_coverage.cram"
bsub -q medium -J NA19238 -o NA19238.log -sla miket_sc "samtools fastq -1 NA19238.1.fq -2 NA19238.2.fq NA19238.alt_bwamem_GRCh38DH.20150715.YRI.high_coverage.cram"
bsub -q medium -J NA19239 -o NA19239.log -sla miket_sc "samtools fastq -1 NA19239.1.fq -2 NA19239.2.fq NA19239.alt_bwamem_GRCh38DH.20150715.YRI.high_coverage.cram"
bsub -q medium -J NA19240 -o NA19240.log -sla miket_sc "samtools fastq -1 NA19240.1.fq -2 NA19240.2.fq NA19240.alt_bwamem_GRCh38DH.20150715.YRI.high_coverage.cram"
```

## Step2: downsample fastq to 30X
 - downsample read 1 from 75X to 30X:
```
bsub -q normal  -J HG00513 -o HG00513.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  HG00513.1.fq .4 > HG00513.1.30X_S100.fq"
bsub -q normal  -J HG00514 -o HG00514.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  HG00514.1.fq .4 > HG00514.1.30X_S100.fq"  
bsub -q normal  -J HG00731 -o HG00731.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  HG00731.1.fq .4 > HG00731.1.30X_S100.fq"  
bsub -q normal  -J HG00732 -o HG00732.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  HG00732.1.fq .4 > HG00732.1.30X_S100.fq"  
bsub -q normal  -J HG00733 -o HG00733.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  HG00733.1.fq .4 > HG00733.1.30X_S100.fq"  
bsub -q normal  -J NA19238 -o NA19238.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  NA19238.1.fq .4 > NA19238.1.30X_S100.fq"  
bsub -q normal  -J NA19239 -o NA19239.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  NA19239.1.fq .4 > NA19239.1.30X_S100.fq"  
bsub -q normal  -J NA19240 -o NA19240.1.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk sample -s100  NA19240.1.fq .4 > NA19240.1.30X_S100.fq"  
```
 - extract read names from read 1
```
bsub -q normal -J HG00512 -o HG00512.1b.log -sla miket_sc "grep '@'  HG00512.1.30X_S100.fq > HG00512.30X_S100.names"
bsub -q normal -J HG00513 -o HG00513.1b.log -sla miket_sc "grep '@'  HG00513.1.30X_S100.fq > HG00513.30X_S100.names" 
bsub -q normal -J HG00514 -o HG00514.1b.log -sla miket_sc "grep '@'  HG00514.1.30X_S100.fq > HG00514.30X_S100.names" 
bsub -q normal -J HG00731 -o HG00731.1b.log -sla miket_sc "grep '@'  HG00731.1.30X_S100.fq > HG00731.30X_S100.names" 
bsub -q normal -J HG00732 -o HG00732.1b.log -sla miket_sc "grep '@'  HG00732.1.30X_S100.fq > HG00732.30X_S100.names" 
bsub -q normal -J HG00733 -o HG00733.1b.log -sla miket_sc "grep '@'  HG00733.1.30X_S100.fq > HG00733.30X_S100.names" 
bsub -q normal -J NA19238 -o NA19238.1b.log -sla miket_sc "grep '@'  NA19238.1.30X_S100.fq > NA19238.30X_S100.names" 
bsub -q normal -J NA19239 -o NA19239.1b.log -sla miket_sc "grep '@'  NA19239.1.30X_S100.fq > NA19239.30X_S100.names" 
bsub -q normal -J NA19240 -o NA19240.1b.log -sla miket_sc "grep '@'  NA19240.1.30X_S100.fq > NA19240.30X_S100.names" 

bsub -q short -J  HG00512 -o HG00512.1c.log -sla miket_sc  "sed -e 's/@//'  HG00512.30X_S100.names > HG00512.30X_S100.names_2" 
bsub -q short -J  HG00513 -o HG00513.1c.log -sla miket_sc  "sed -e 's/@//'  HG00513.30X_S100.names > HG00513.30X_S100.names_2" 
bsub -q short -J  HG00514 -o HG00514.1c.log -sla miket_sc  "sed -e 's/@//'  HG00514.30X_S100.names > HG00514.30X_S100.names_2" 
bsub -q short -J  HG00731 -o HG00731.1c.log -sla miket_sc  "sed -e 's/@//'  HG00731.30X_S100.names > HG00731.30X_S100.names_2" 
bsub -q short -J  HG00732 -o HG00732.1c.log -sla miket_sc  "sed -e 's/@//'  HG00732.30X_S100.names > HG00732.30X_S100.names_2" 
bsub -q short -J  HG00733 -o HG00733.1c.log -sla miket_sc  "sed -e 's/@//'  HG00733.30X_S100.names > HG00733.30X_S100.names_2" 
bsub -q short -J  NA19238 -o NA19238.1c.log -sla miket_sc  "sed -e 's/@//'  NA19238.30X_S100.names > NA19238.30X_S100.names_2" 
bsub -q short -J  NA19239 -o NA19239.1c.log -sla miket_sc  "sed -e 's/@//'  NA19239.30X_S100.names > NA19239.30X_S100.names_2" 
bsub -q short -J  NA19240 -o NA19240.1c.log -sla miket_sc  "sed -e 's/@//'  NA19240.30X_S100.names > NA19240.30X_S100.names_2" 
```
 - extract reads with the same name from read2
```
bsub -q big -R 'rusage[mem=36000]' -J HG00512 -o HG00512.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00512.2.fq HG00512.30X_S100.names_2 > HG00512.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J HG00513 -o HG00513.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00513.2.fq HG00513.30X_S100.names_2 > HG00513.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J HG00514 -o HG00514.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00514.2.fq HG00514.30X_S100.names_2 > HG00514.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J HG00731 -o HG00731.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00731.2.fq HG00731.30X_S100.names_2 > HG00731.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J HG00732 -o HG00732.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00732.2.fq HG00732.30X_S100.names_2 > HG00732.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J HG00733 -o HG00733.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq HG00733.2.fq HG00733.30X_S100.names_2 > HG00733.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J NA19238 -o NA19238.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq NA19238.2.fq NA19238.30X_S100.names_2 > NA19238.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J NA19239 -o NA19239.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq NA19239.2.fq NA19239.30X_S100.names_2 > NA19239.2.30X_S100.fq"
bsub -q big -R 'rusage[mem=36000]' -J NA19240 -o NA19240.2.fq.log -sla miket_sc "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq NA19240.2.fq NA19240.30X_S100.names_2 > NA19240.2.30X_S100.fq"
```

## Step3: split fastq to sub-fastq for faster alignment
```
split -l 8000000 HG00512.30X_S100.names_2  HG00512.30X_S100.names_split.
split -l 8000000 HG00513.30X_S100.names_2  HG00513.30X_S100.names_split.
split -l 8000000 HG00514.30X_S100.names_2  HG00514.30X_S100.names_split.
split -l 8000000 HG00731.30X_S100.names_2  HG00731.30X_S100.names_split.
split -l 8000000 HG00732.30X_S100.names_2  HG00732.30X_S100.names_split.
split -l 8000000 HG00733.30X_S100.names_2  HG00733.30X_S100.names_split.
split -l 8000000 NA19238.30X_S100.names_2  NA19238.30X_S100.names_split.
split -l 8000000 NA19239.30X_S100.names_2  NA19239.30X_S100.names_split.
split -l 8000000 NA19240.30X_S100.names_2  NA19240.30X_S100.names_split.
```
	- match read pairs and name-sort each sub-fastq
```
import os
for k1 in os.listdir('downsamples/'):
 if k1.split('.')[-2]=='names_split':
  sample_name=k1.split('.')[0]
  fq1_name=sample_name+'.1.30X_S100.fq'
  fq2_name=sample_name+'.2.30X_S100.fq'
  os.system(r'''bsub -q medium -J %s -o %s -sla miket_sc -sp 100 "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq %s %s > %s" '''%(k1+'.1.fq','logs_downsample/split_fq.'+k1+'.1.log', fq1_name, 'downsamples/'+k1, k1+'.1.fq'))
  os.system(r'''bsub -q medium -J %s -o %s -sla miket_sc -sp 100 "/data/talkowski/xuefang/local/src/seqtk/seqtk subseq %s %s > %s" '''%(k1+'.1.fq','logs_downsample/split_fq.'+k1+'.2.log', fq2_name, 'downsamples/'+k1, k1+'.2.fq'))

import os
for k1 in os.listdir('.'):
 if not 'names_split' in k1:continue
 if k1.split('.')[-1]=='fq' and k1.split('.')[-4]=='names_split':
  os.system(r'''bsub -q short -J %s -o %s -sla miket_sc -sp 100 "bgzip %s" '''%(k1, k1+'.log', k1))
```

	- sort fastq by names
```
import os
for k1 in os.listdir('.'): 
 if k1.split('.')[-1]=='gz':
  os.system(r''' bsub -q short -J %s -o %s -sla miket_sc -sp 100 " zcat %s | paste - - - - | sort -k1,1 -S 3G | tr '\t' '\n' > %s"'''
   %(k1, k1+'.log',k1, 'sorted_fq/'+k1.replace('.fq.gz','.sorted.fq')))
```

## Step4: align sub bams 
```
ref='/data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa'
import os
for k1 in os.listdir('sorted_fq/'):
 if k1.split('.')[-1]=='fq' and k1.split('.')[-2]=='sorted' and k1.split('.')[-3]=='1':
  os.system(r'''bsub -q normal -J %s -o %s -sp 100 -sla miket_sc "bwa mem -t 1 -B 4 -O 6 -E 1 -M %s %s %s | samtools view -1 - > %s" '''%(k1, k1.replace('.sorted.fq','.sorted.log'),ref, 'sorted_fq/'+k1,'sorted_fq/'+k1.replace('.1.sorted.fq','.2.sorted.fq'), 'alignment_1/'+k1.replace('.1.sorted.fq','.bam')))
  print(k1)
```

 - sort sub bams 
```
workdir='/data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_GRCh38/fastq/alignment_1/'
import os
for k1 in os.listdir('.'):
 if k1.split('.')[-1]=='bam':
  os.system(r'''bsub -q medium -J %s -o %s -sp 100 -sla miket_sc "samtools sort -o %s -T %s %s" '''%(k1.replace('.bam',''), k1.replace('.bam','sort.log'), k1.replace('.bam','.sorted.bam'),'tmp/'+k1.replace('.bam',''),k1 ))
```

 - merge sub bams 

```
import os
samp_hash={}
for k1 in os.listdir('.'):
 if k1.split('.')[-1]=='bam' and k1.split('.')[-2]=='sorted':
  if not k1.split('.')[0] in samp_hash.keys():
   samp_hash[k1.split('.')[0]]=[[]]
  if len(samp_hash[k1.split('.')[0]][-1])<7:
   samp_hash[k1.split('.')[0]][-1].append(k1)
  else:
   samp_hash[k1.split('.')[0]].append([k1])

for k1 in samp_hash.keys():
 rec=0
 for k2 in samp_hash[k1]:
  rec+=1
  fileout=k1+'.'+str(rec)+'.bam'
  if len(k2)==7:
   os.system(r'''bsub -q medium -J %s -o %s -sp 100 -sla miket_sc "samtools merge %s %s %s %s %s %s %s %s" '''%(k1+'.'+str(rec),k1+'.'+str(rec)+'.merge1.log',fileout, k2[0], k2[1], k2[2], k2[3], k2[4], k2[5], k2[6]))
  else:
   print(' '.join(['bsub -q medium -J',k1+'.'+str(rec), '-o', k1+'.'+str(rec)+'.merge1.log' , '-sp 100 -sla miket_sc', 'samtools merge', fileout, ' '.join(k2) ]))
```

 - merge sub bams
```
workdir: /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_GRCh38/fastq/alignment_1

import os
samp_hash={}
for k1 in os.listdir('.'):
 if k1.split('.')[-1]=='bam' and '.RG.' in k1:
  if not k1.split('.')[0] in samp_hash.keys():
   samp_hash[k1.split('.')[0]]=[]
  samp_hash[k1.split('.')[0]].append(k1)

for k1 in samp_hash.keys():
 samp_hash[k1].sort()

for k1 in samp_hash.keys():
 k2=samp_hash[k1]
 fileout=k1+'.with_RG.sorted.bam'
 os.system(r'''bsub -q medium -J %s -o %s -sp 100 -sla miket_sc "samtools merge %s %s %s %s %s %s %s %s" '''%(k1,k1+'.merge2.log',fileout, k2[0], k2[1], k2[2], k2[3], k2[4], k2[5], k2[6]))
```

 - index bams
```
workdir: /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_GRCh38/fastq/alignment_1

for k1 in os.listdir(): 
 if k1.split('.')[-1]=='bam' and k1.split('.')[-2]=='sorted':
  os.system(r'''bsub -q medium -J %s -o %s -sp 100 -sla miket_sc "samtools index %s" '''%(k1,k1.replace('.bam','.index.log'), k1))
```


## Step5: add readgroups (note: this is not an necessary step, read group should have been included during alignent)
```
bsub -q big -R 'rusage[mem=36000]' -J HG00512 -o HG00512.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00512.sorted.bam O=HG00512.RG1.bam RGID=HG00512 RGLB=HG00512 RGPL=ILLUMINA RGPU=HG00512 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J HG00513 -o HG00513.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00513.sorted.bam O=HG00513.RG1.bam RGID=HG00513 RGLB=HG00513 RGPL=ILLUMINA RGPU=HG00513 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J HG00514 -o HG00514.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00514.sorted.bam O=HG00514.RG1.bam RGID=HG00514 RGLB=HG00514 RGPL=ILLUMINA RGPU=HG00514 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J HG00731 -o HG00731.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00731.sorted.bam O=HG00731.RG1.bam RGID=HG00731 RGLB=HG00731 RGPL=ILLUMINA RGPU=HG00731 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J HG00732 -o HG00732.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00732.sorted.bam O=HG00732.RG1.bam RGID=HG00732 RGLB=HG00732 RGPL=ILLUMINA RGPU=HG00732 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J HG00733 -o HG00733.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=HG00733.sorted.bam O=HG00733.RG1.bam RGID=HG00733 RGLB=HG00733 RGPL=ILLUMINA RGPU=HG00733 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J NA19238 -o NA19238.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=NA19238.sorted.bam O=NA19238.RG1.bam RGID=NA19238 RGLB=NA19238 RGPL=ILLUMINA RGPU=NA19238 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J NA19239 -o NA19239.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=NA19239.sorted.bam O=NA19239.RG1.bam RGID=NA19239 RGLB=NA19239 RGPL=ILLUMINA RGPU=NA19239 RGSM=20"
bsub -q big -R 'rusage[mem=36000]' -J NA19240 -o NA19240.addRG.log -sla miket_sc -sp 100 "java -Xmx32G -jar /data/talkowski/xuefang/local/src/picard/build/libs/picard.jar AddOrReplaceReadGroups I=NA19240.sorted.bam O=NA19240.RG1.bam RGID=NA19240 RGLB=NA19240 RGPL=ILLUMINA RGPU=NA19240 RGSM=20"


bsub -q medium -J HG00512 -o HG00512.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00512.RG1.bam"
bsub -q medium -J HG00513 -o HG00513.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00513.RG1.bam"
bsub -q medium -J HG00514 -o HG00514.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00514.RG1.bam"
bsub -q medium -J HG00731 -o HG00731.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00731.RG1.bam"
bsub -q medium -J HG00732 -o HG00732.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00732.RG1.bam"
bsub -q medium -J HG00733 -o HG00733.RG1.index.log -sp 100 -sla miket_sc "samtools index HG00733.RG1.bam"
bsub -q medium -J NA19238 -o NA19238.RG1.index.log -sp 100 -sla miket_sc "samtools index NA19238.RG1.bam"
bsub -q medium -J NA19239 -o NA19239.RG1.index.log -sp 100 -sla miket_sc "samtools index NA19239.RG1.bam"
bsub -q medium -J NA19240 -o NA19240.RG1.index.log -sp 100 -sla miket_sc "samtools index NA19240.RG1.bam"
```

### Step5a: calculate SNP recalibrate table:
```
bsub -q big -R 'rusage[mem=36000]' -J HG00512.recali.SNP -o HG00512.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00512.recal_data.table -I HG00512.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J HG00513.recali.SNP -o HG00513.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00513.recal_data.table -I HG00513.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J HG00514.recali.SNP -o HG00514.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00514.recal_data.table -I HG00514.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J HG00731.recali.SNP -o HG00731.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00731.recal_data.table -I HG00731.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J HG00732.recali.SNP -o HG00732.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00732.recal_data.table -I HG00732.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J HG00733.recali.SNP -o HG00733.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o HG00733.recal_data.table -I HG00733.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J NA19238.recali.SNP -o NA19238.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o NA19238.recal_data.table -I NA19238.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J NA19239.recali.SNP -o NA19239.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o NA19239.recal_data.table -I NA19239.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
bsub -q big -R 'rusage[mem=36000]' -J NA19240.recali.SNP -o NA19240.recali.SNP.log -sla miket_sc -sp 100 "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T BaseRecalibrator -nt 1 -l INFO -cov ReadGroupCovariate -cov QualityScoreCovariate -cov CycleCovariate -cov ContextCovariate -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa -o NA19240.recal_data.table -I NA19240.RG1.bam -knownSites /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/other_mapping_resources/ALL.wgs.dbsnp.build135.snps.sites.vcf.gz"
```
### Step5b: produce bams with base quality recalibrated
```
bsub -q big -R 'rusage[mem=36000]' -J HG00512.recali.SNP -o HG00512.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00512.recali.bam -I HG00512.RG1.bam -BQSR HG00512.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J HG00513.recali.SNP -o HG00513.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00513.recali.bam -I HG00513.RG1.bam -BQSR HG00513.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J HG00514.recali.SNP -o HG00514.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00514.recali.bam -I HG00514.RG1.bam -BQSR HG00514.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J HG00731.recali.SNP -o HG00731.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00731.recali.bam -I HG00731.RG1.bam -BQSR HG00731.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J HG00732.recali.SNP -o HG00732.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00732.recali.bam -I HG00732.RG1.bam -BQSR HG00732.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J HG00733.recali.SNP -o HG00733.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o HG00733.recali.bam -I HG00733.RG1.bam -BQSR HG00733.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J NA19238.recali.SNP -o NA19238.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o NA19238.recali.bam -I NA19238.RG1.bam -BQSR NA19238.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J NA19239.recali.SNP -o NA19239.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o NA19239.recali.bam -I NA19239.RG1.bam -BQSR NA19239.recal_data.table --disable_bam_indexing" 
bsub -q big -R 'rusage[mem=36000]' -J NA19240.recali.SNP -o NA19240.recali2.SNP.log -sla miket_sc -sp 100  "java -Xmx32G -jar /PHShome/hw878/Software/GATK/GenomeAnalysisTK.jar -T PrintReads -l INFO  -R /data/talkowski/Samples/1000Genomes/HGSV_Illumina_Alignment_hs37d5/reference/hs37d5.fa  -o NA19240.recali.bam -I NA19240.RG1.bam -BQSR NA19240.recal_data.table --disable_bam_indexing" 
```

### Step5c: Mard duplicates in the alignment with biobambam2
```
bsub -q big -R 'rusage[mem=36000]' -J HG00512.markdup.SNP -o HG00512.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00512.recali.bam O=HG00512.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J HG00513.markdup.SNP -o HG00513.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00513.recali.bam O=HG00513.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J HG00514.markdup.SNP -o HG00514.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00514.recali.bam O=HG00514.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J HG00731.markdup.SNP -o HG00731.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00731.recali.bam O=HG00731.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J HG00732.markdup.SNP -o HG00732.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00732.recali.bam O=HG00732.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J HG00733.markdup.SNP -o HG00733.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=HG00733.recali.bam O=HG00733.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J NA19238.markdup.SNP -o NA19238.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=NA19238.recali.bam O=NA19238.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J NA19239.markdup.SNP -o NA19239.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=NA19239.recali.bam O=NA19239.recali.markdup.bam index=1 rmdup=0"
bsub -q big -R 'rusage[mem=36000]' -J NA19240.markdup.SNP -o NA19240.markdup.log -sla miket_sc -sp 100  "/data/talkowski/xuefang/local/src/biobambam2/bin/bammarkduplicates I=NA19240.recali.bam O=NA19240.recali.markdup.bam index=1 rmdup=0"
```




